### DISCUSSION TITLE
Add "Clean-param" directive in robots.txt

### DISCUSSION THREAD

**Discussion by ruslanyar** [vercel/next.js#62872] [Ideas]
### Goals

Include "Clean-param" directive to resolve by resolveRobots

### Background

I am facing with problem that when generating the robots.txt file, the ["clean-param" directive](https://yandex.ru/support/webmaster/robot-workings/clean-param.html?lang=en) is ignored.
Yandex recommends using "Clean-param" for cleaning URLs contains GET parameters because this directive allows transfer some accumulated metrics to the main URL or website.

### Proposal

Add `cleanParam?` to the RobotsFile type and update the function that handles robots.txt rules.
I am interested in contributing.

**Comment by gormonn** [comment]
Any workarounds?

**Reply by ruslanyar** [reply]
Unfortunately, I haven't found it yet

**Comment by AhmadReshadarm** [comment]
Here is a way to work around this and it works for me. Hope this help)

`    {
        userAgent: 'Yandex',
        disallow: '\nClean-param: ref /some_dir/get_book.pl',
      }
`
